core/app/models/spree/payment_method/store_credit.rb

```diff
@@ -67,7 +67,7 @@ def credit(amount_in_cents, auth_code, gateway_options = {})
         originator = gateway_options[:originator]
 
         store_credit.credit(
-          amount_in_cents / 100.0.to_d,
+          ::Money.new(amount_in_cents, currency).to_d,
```

Comment:
This is creating a RubyMoney in order to reverse the amount_in_cents logic. Example:

```Ruby
[1] pry(main)> ::Money.new(1000, 'USD').to_d.to_s
=> "10.0"
[2] pry(main)> ::Money.new(1000, 'JPY').to_d.to_s
=> "1000.0"
```